httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt

```diff
@@ -72,6 +75,42 @@ object TbdexHttpClient {
     }
   }
 
+  /**
+   * Fetches balances from a PFI.
+   *
+   * @param pfiDid The decentralized identifier of the PFI.
+   * @param requesterDid The decentralized identifier of the entity requesting the balances.
+   * @return A list of [Balance] matching the request.
+   * @throws TbdexResponseException for request or response errors.
```

Comment:
Thinking out loud: in tbdex-js, we have both `TbdexRequestError` and `TbdexResponseError`, representing errors that happen before and after the client call respectively. It's a slight misnomer to throw a RequestException for both request and response errors, so I'm making a mental note that in my next tidy-up PR, I may explore introducing a `TbdexRequestException` class. In any case, it's relatively low priority and outside the scope of this PR.